### PR TITLE
Allow individual bridges, bundles and components to be used with 4.0

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Symfony/Bridge/Twig/composer.json [hunk 1]
```diff
@@ -21,22 +21,22 @@
...
-        "symfony/console": "~2.8|~3.0",
-        "symfony/var-dumper": "~2.8.10|~3.1.4|~3.2",
-        "symfony/expression-language": "~2.8|~3.0",
-        "symfony/web-link": "~3.3"
+        "symfony/routing": "~2.8|~3.0|~4.0",
+        "symfony/templating": "~2.8|~3.0|~4.0",
+        "symfony/translation": "~2.8|~3.0|~4.0",
+        "symfony/yaml": "~2.8|~3.0|~4.0",
+        "symfony/security": "~2.8|~3.0|~4.0",
+        "symfony/security-acl": "~2.8|~3.0|~4.0",
```
Comment (xabbuh) [inline src/Symfony/Bridge/Twig/composer.json]: Are we sure that we will bump the release for the Security Acl component? I don't think so (we did not really add anything there).

File: src/Symfony/Bundle/SecurityBundle/composer.json [hunk 2]
```diff
@@ -17,29 +17,29 @@
...
-        "symfony/expression-language": "~2.8|~3.0",
+        "symfony/asset": "~2.8|~3.0|~4.0",
+        "symfony/browser-kit": "~2.8|~3.0|~4.0",
+        "symfony/console": "~3.2|~4.0",
+        "symfony/css-selector": "~2.8|~3.0|~4.0",
+        "symfony/dom-crawler": "~2.8|~3.0|~4.0",
+        "symfony/form": "^2.8.18|^3.2.5|~4.0",
+        "symfony/framework-bundle": "^3.2.8|~4.0",
+        "symfony/http-foundation": "~2.8|~3.0|~4.0",
+        "symfony/security-acl": "~2.8|~3.0|~4.0",
```
Comment (xabbuh) [inline src/Symfony/Bundle/SecurityBundle/composer.json]: see my comment above

File: src/Symfony/Component/HttpKernel/composer.json [hunk 3]
```diff
@@ -17,27 +17,27 @@
...
-        "symfony/expression-language": "~2.8|~3.0",
-        "symfony/finder": "~2.8|~3.0",
-        "symfony/process": "~2.8|~3.0",
-        "symfony/routing": "~2.8|~3.0",
-        "symfony/stopwatch": "~2.8|~3.0",
-        "symfony/templating": "~2.8|~3.0",
-        "symfony/translation": "~2.8|~3.0",
-        "symfony/var-dumper": "~3.3",
+        "symfony/browser-kit": "~2.8|~3.0|~4.0",
+        "symfony/class-loader": "~2.8|~3.0|~4.0",
```
Comment (xabbuh) [inline src/Symfony/Component/HttpKernel/composer.json]: the ClassLoader component will be removed with Symfony 4

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 4]
```diff
@@ -18,42 +18,42 @@
...
         "ext-xml": "*",
-        "symfony/cache": "~3.3",
-        "symfony/class-loader": "~3.2",
-        "symfony/dependency-injection": "~3.3-beta2",
-        "symfony/config": "~3.3",
-        "symfony/event-dispatcher": "~3.3",
-        "symfony/http-foundation": "~3.3",
-        "symfony/http-kernel": "~3.3",
+        "symfony/cache": "~3.3|~4.0",
+        "symfony/class-loader": "~3.2|~4.0",
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/composer.json]: see my comment above

**Review by xabbuh — CHANGES_REQUESTED** [review]
This should be done on the `3.4` branch.

**Review by xabbuh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
On 3.4, we should allow only 4.0.0

File: src/Symfony/Bridge/ProxyManager/composer.json [hunk 5]
```diff
@@ -17,11 +17,11 @@
     ],
     "require": {
         "php": ">=5.5.9",
-        "symfony/dependency-injection": "~2.8|~3.0",
-        "ocramius/proxy-manager": "~0.4|~1.0|~2.0"
+        "symfony/dependency-injection": "~2.8|~3.0|~4.0",
+        "ocramius/proxy-manager": "~0.4|~1.0|~2.0|~4.0"
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/ProxyManager/composer.json]: not sure about this one

File: src/Symfony/Bundle/WebServerBundle/composer.json [hunk 6]
```diff
@@ -17,9 +17,9 @@
     ],
     "require": {
         "php": ">=5.5.9",
-        "symfony/console": "~2.8.8|~3.0.8|~3.1.2|~3.2",
-        "symfony/http-kernel": "~2.8|~3.0",
-        "symfony/process": "~2.8|~3.0"
+        "symfony/console": "~2.8.8|~3.0.8|~3.1.2|~3.2|~4.0",
+        "symfony/http-kernel": "~2.8|~3.|~4.00",
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/WebServerBundle/composer.json]: typo

**Comment by dunglas** [comment]
Done, to do the merge in 4.0:

```
find src -name "composer.json" -exec sed -i '' 's/\|\~4\.0\.0"/\|\~4\.0"/g' {} \;
find src -name "composer.json" -exec sed -i '' 's/\|\^4\.0\.0"/\|\^4\.0"/g' {} \;
```

(I can handle the merge if can help)

File: src/Symfony/Component/Dotenv/composer.json [hunk 7]
```diff
@@ -19,7 +19,7 @@
         "php": ">=5.5.9"
     },
     "require-dev": {
-        "symfony/process": "^3.2"
+        "symfony/process": "^3.2|^4.0.0"
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Dotenv/composer.json]: all ^4.0.0 should be replaced by ~4.0.0

**Comment by nicolas-grekas** [comment]
Thank you @dunglas.

**Pull request closed without merging**